File: pkg/java/src/main/java/dev/openfga/language/validation/ValidationRegex.java
package dev.openfga.language.validation;

import java.util.regex.Pattern;
import java.util.regex.PatternSyntaxException;

class ValidationRegex {

    private final String rule;
    private final Pattern regex;

    private ValidationRegex(String rule, Pattern regex) {
        this.rule = rule;
        this.regex = regex;
    }

    public String getRule() {
        return rule;
    }

    public Pattern getRegex() {
        return regex;
    }

    public static ValidationRegex build(String name, String rule) {
        Pattern regex = null;
        try {
            regex = Pattern.compile(rule);
        } catch (PatternSyntaxException e) {
            var message = "Incorrect " + name + " regex specification for  + rule";
            throw new IllegalArgumentException(message);
        }
        return new ValidationRegex(rule, regex);
    }

    public boolean matches(String input) {
        return regex.matcher(input).matches();
    }
}

File: pkg/csharp/src/OpenFga.Language/Validation/ValidationRegex.cs
using System.Text.RegularExpressions;

namespace OpenFga.Language.Validation;

public class ValidationRegex {
    private readonly string _rule;
    private readonly Regex _regex;

    private ValidationRegex(string rule, Regex regex) {
        _rule = rule;
        _regex = regex;
    }

    public string Rule => _rule;
    public Regex Regex => _regex;

    public static ValidationRegex Build(string name, string rule) {
        Regex regex;
        try {
            regex = new Regex(rule);
        }
        catch (ArgumentException ex) {
            var message = $"Incorrect {name} regex specification for {rule}";
            throw new ArgumentException(message, ex);
        }
        return new ValidationRegex(rule, regex);
    }

    public bool Matches(string input) {
        return _regex.IsMatch(input);
    }
}